src/Chapter2.hs

```diff
@@ -842,7 +872,12 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+--rotate = error "rotate: Not implemented!"
+rotate :: (Num a) => Int -> [a] -> [a]
+rotate n xs 
+    | n < 0 = []
+    | otherwise = drop n $ take (length xs + n) $ cycle xs
```

Comment:
Absolutely correct!

You can optimise this solution if you use `mod` by the length of the list on `n`, so you won't do extra work on big input numbers 🙂